### PR TITLE
Playbook: Email Address Enrichment - Generic

### PR DESCRIPTION
**Pull request by JonathanMeler**
- GetStringDistance
- EmailDomainSquattingReputation

email_address_enrichment_-_generic_thu_jan_18_2018

fixes https://github.com/demisto/etc/issues/10313

### CODE REVIEW

**Comment by asafshen** [comment]
@JonathanMeler 
we have made some changes in playbook structure - please download the playbook again from an updated demisto env (demo2 for example)

**Comment by JonathanMeler** [comment]
Test demo run: https://demo2.demisto.com/#/WorkPlan/37812

File: Scripts/script-EmailDomainSquattingReputation.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  id: EmailDomainSquattingReputation
+  version: -1
+name: EmailDomainSquattingReputation
+script: |2-
+
+  var email = args.email;
+  var domains = argToList(args.domain);
+  var threshold = parseInt(args.threshold);
+
+  var emailParts = email.split('@',2);
```
Comment (anara123) [inline Scripts/script-EmailDomainSquattingReputation.yml]: We use python convention. emailParts => email_parts

Comment (anara123) [reply Scripts/script-EmailDomainSquattingReputation.yml]: change in all scripts

File: Scripts/script-EmailDomainSquattingReputation.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  if (emailParts.length < 2){
+      return { ContentsFormat: formats.text, Type: entryTypes.error, Contents: email.toString() + " - is not a valid email address" };
+  }
+  var emailObj = {Username: emailParts[0],
```
Comment (anara123) [inline Scripts/script-EmailDomainSquattingReputation.yml]: indent and format json objects properly

File: Scripts/script-EmailDomainSquattingReputation.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+script: |2-
+
+  var email = args.email;
+  var domains = argToList(args.domain);
+  var threshold = parseInt(args.threshold);
+
+  var emailParts = email.split('@',2);
+
+  if (emailParts.length < 2){
+      return { ContentsFormat: formats.text, Type: entryTypes.error, Contents: email.toString() + " - is not a valid email address" };
```
Comment (anara123) [inline Scripts/script-EmailDomainSquattingReputation.yml]: indent and format json objects properly

File: Scripts/script-EmailDomainSquattingReputation.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      });
+  });
+  var ec = {};
+  var suspicious = dq(emailObj,"Distance(val.Value > 0 && val.Value < {0}).Value".format(threshold));
+  var dbotScore = 0;
+  var malicious = null;
+
+  if(suspicious && suspicious.length > 0){
+      //add dbot score, suspicious
+      ec.DBotScore = {Indicator: email, Type: 'email', Vendor: 'DomainSquatting', Score: 2};
```
Comment (anara123) [inline Scripts/script-EmailDomainSquattingReputation.yml]: format json

File: Scripts/script-EmailDomainSquattingReputation.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+      return { ContentsFormat: formats.text, Type: entryTypes.error, Contents: email.toString() + " - is not a valid email address" };
+  }
+  var emailObj = {Username: emailParts[0],
+                  Domain: emailParts[1],
+                  Address : email,
+                  Distance : []
+  };
+
+  domains.forEach(function(domain){
+      var resp = executeCommand("GetStringsDistance", {inputString: emailObj.Domain, compareString: domain});
```
Comment (anara123) [inline Scripts/script-EmailDomainSquattingReputation.yml]: format and indent object

File: Scripts/script-GetStringsDistance.yml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                              'TooClose' : levenshteinDistance > 0 and levenshteinDistance < closeDistanceInt
+                              })
+          res.append({'Type' : entryTypes['note'],
+                          'Contents': {'Distances' : distances},
+                          'ContentsFormat' : formats['json'],
+                          'HumanReadable': tblToMd('Distances', distances, ['StringA','StringB','LevenshteinDistance','TooClose']),
+                          'ReadableContentsFormat' : formats['markdown']})
+
+      else:
+          res.append({'Type': entryTypes['error'], 'ContentsFormat': formats['text'], 'Contents': 'Unable to extract inputString - ' + inputString})
```
Comment (anara123) [inline Scripts/script-GetStringsDistance.yml]: format

File: Scripts/script-GetStringsDistance.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      if inputString:
+          distances = []
+          for curString in compareString:
+              levenshteinDistance = levenshtein(curString, inputString)
+              distances.append({'StringA' : inputString,
+                              'StringB' : curString,
+                              'LevenshteinDistance' : levenshteinDistance,
+                              'TooClose' : levenshteinDistance > 0 and levenshteinDistance < closeDistanceInt
+                              })
+          res.append({'Type' : entryTypes['note'],
```
Comment (anara123) [inline Scripts/script-GetStringsDistance.yml]: format

File: Scripts/script-GetStringsDistance.yml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  compareString = argToList(demisto.get(demisto.args(), 'compareString'))
+  if not compareString:
+      res.append({'Type': entryTypes['error'], 'ContentsFormat': formats['text'], 'Contents': 'Unable to extract compareString from arguments'})
+  else:
+      inputString = demisto.get(demisto.args(), 'inputString')
+      if inputString:
+          distances = []
+          for curString in compareString:
+              levenshteinDistance = levenshtein(curString, inputString)
+              distances.append({'StringA' : inputString,
```
Comment (anara123) [inline Scripts/script-GetStringsDistance.yml]: format

File: Scripts/script-IsEmailAddressInternal.yml [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+              "Email": {
+                  "Address": email,
+                  "Domain" : parts[1],
+                  "Username": parts[0],
+                  "NetworkType" : networkType
+              }
+          }
+      }
+      demisto.results({ "Type" : entryTypes["note"],
+                      "ContentsFormat" : formats["json"],
```
Comment (anara123) [inline Scripts/script-IsEmailAddressInternal.yml]: format

File: TestPlaybooks/playbook-Test_-_Email_Address_Enrichment_-_Generic.yml [hunk 10]
```diff
@@ -0,0 +1,214 @@
+id: test_-_email_address_enrichment_-_generic
+name: Test - Email Address Enrichment - Generic
```
Comment (anara123) [inline TestPlaybooks/playbook-Test_-_Email_Address_Enrichment_-_Generic.yml]: change the name to be Email Address Enrichment - Generic - Test

**Pull request merged**